llvm/include/llvm/Transforms/IPO/Attributor.h

```diff
@@ -6106,6 +6106,56 @@ struct AAPointerInfo : public AbstractAttribute {
     Type *Ty;
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 5754]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Since you have pure virtual function, I suggest to put the constructor in `protect`.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/72029?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Apologies, I am having trouble connecting to Graphite. 
Also, which virtual function are you taking about?

Reply:
Okay, that line is far above. It is the c'tor of `AAPointerInfo`.

Reply:
do you mean 

```
struct AAPointerInfo : public AbstractAttribute {
  protected:
    AAPointerInfo(const IRPosition &IRP) : AbstractAttribute(IRP) {}
```

This would mean that I can't use AApointerinfo in AAallocationinfo unless I inherit from AApointerinfo in allocation info?